src/RelayVariablesInAllowedPosition.ts

```diff
@@ -0,0 +1,115 @@
+import { GraphQLSchema, GraphQLType, ValueNode, ValidationRule } from "graphql"
+import { isNonNullType, isTypeSubTypeOf, typeFromAST, GraphQLError, parseType } from "./dependencies"
+import { getArgumentDefinitions } from "./argumentDefinitions"
+
+export const RelayVariablesInAllowedPosition: ValidationRule = function RelayVariablesInAllowedPosition(context) {
+  let varDefMap = Object.create(null)
+  return {
+    FragmentDefinition: {
+      enter(fragmentDefinition) {
+        varDefMap = Object.create(null)
+        const argumentDefinitions = getArgumentDefinitions(fragmentDefinition)
+
+        if (argumentDefinitions != null) {
+          argumentDefinitions.forEach(def => {
+            if (def.value.kind !== "ObjectValue") {
+              return
+            }
+            const typeField = def.value.fields.find(f => f.name.value === "type")
+            if (typeField == null || typeField.value.kind !== "StringValue") {
+              return
+            }
+            try {
+              const parsedType = parseType(typeField.value.value)
+              varDefMap[def.name.value] = {
+                type: parsedType,
+                node: def,
+              }
+            } catch (e) {}
```

Comment:
Perhaps it could be helpful to log this?

Reply:
I believe it is supposed to be caught and handled by a different validation rule?

Reply:
Ah ok, that works for me then 👍